demos/ios/MASVS-CRYPTO/MASTG-DEMO-0073/MastgTest.swift

```diff
@@ -0,0 +1,170 @@
+import Foundation
+import Security
+import Darwin
+import CommonCrypto
+
+// Unsafe bindings to libc srand and rand
+@_silgen_name("srand")
+func c_srand(_ seed: UInt32)
+
+@_silgen_name("rand")
+func c_rand() -> Int32
+
+struct MastgTest {
+
+    // Insecure: libc rand seeded with time, predictable and not suitable for cryptography
+    static func generateInsecureRandomTokenRand() -> String {
+        var token = ""
+
+        for _ in 0..<16 {
+            let value = c_rand() % 256
+            token += String(format: "%02x", value)
+        }
+        return token
+    }
+
+    // Cryptographically secure on Apple platforms
+    // Swift random APIs use SystemRandomNumberGenerator backed by the system CSPRNG via arc4random_buf
+    // Shown here as a secure source that is not a dedicated crypto token API
+    static func generateInsecureRandomTokenSwiftRandom() -> String {
+        var token = ""
+        for _ in 0..<16 {
+            let b = UInt8.random(in: 0...255)
+            token += String(format: "%02x", b)
+        }
+        return token
+    }
+    
+    // Cryptographically secure: direct read from /dev/random on Apple platforms
+    // However, this is a low level interface and is discouraged in favor of SecRandomCopyBytes
+    static func generateInsecureRandomTokenDevRandom() -> String {
+        let count = 16
+
+        let fd = open("/dev/random", O_RDONLY)
+        if fd < 0 {
+            return "Error opening /dev/random"
+        }
+
+        var buffer = [UInt8](repeating: 0, count: count)
+        let readCount = read(fd, &buffer, count)
+        close(fd)
+
+        if readCount != count {
+            return "Error reading /dev/random"
+        }
+
+        return buffer.map { String(format: "%02x", $0) }.joined()
+    }
+    
+    // Cryptographically secure but discouraged as a direct token API
```

Comment:
```suggestion
    // Cryptographically secure but discouraged as a direct token API in Swift code 
    // because uses legacy C style interfaces that are easier to misuse
```